src/metabase/integrations/ldap.clj

```diff
@@ -121,13 +125,13 @@
       set))
 
 (defn- get-user-groups
-  "Retrieve groups for a supplied DN."
-  ([dn]
-    (with-connection get-user-groups dn))
-  ([conn dn]
+   "Retrieve groups for a supplied DN or UID"
+  ([dn uid]
+    (with-connection get-user-groups dn uid))
+  ([conn dn uid]
     (when (ldap-group-base)
       (let [results (ldap/search conn (ldap-group-base) {:scope      :sub
-                                                         :filter     (str "member=" (escape-value dn))
+                                                         :filter     (str "(|(member=" (escape-value dn) ")(memberUid=" (escape-value uid) "))")
```

Comment:
I don't know a lot about LDAP and how people apply it with groups. Is it common/valuable to use both `member` and `memberUid` in the same install? Any opinions on this @wwwiiilll?

Reply:
Not really. Usually or you either use the memberUid (nis or rfc2307 schema) or the member (rf2307bis schema) way. I have seen servers with both ways, tho. For the 'memberUid' attribute you need to specify the 'uid' of the user. For the 'member' attribute you need to specify the full DN. That's the reason why there's an 'or' on the query

Reply:
Both RFC 2307 and  RFC 2307bis (which is still a draft technically) mention the use of `memberUid` only.

I see no reason to support both at the same time, and acutally, I thin it may have unexpected results if the server does have both. We should allow the admin to select the "key" and the "member" attribute to use, just like we do with the name and e-mail, instead.

Reply:
Also, regarding `memberOf`, it's a special calculated attribute that Active Directory (and probably others) add on the user object to list which groups he's part of.

Reply:
Why would it have unexpected results? To check the group membership is either by checking the memberUid=uid (rfc2307) or member=DN (rfc2307bis). The approach metabase is using right now for user membership is done by using the member attribute only, this is an enhancement.

If this can be done automatically by an OR query, why would you want the admin to select what attribute to use?

Regarding the memberOf, yes, but it is not something found on a OpenLDAP until you do some special 
[tweaks](https://www.adimian.com/blog/2014/10/how-to-enable-memberof-using-openldap/).